src/types.ts

```diff
@@ -935,11 +944,19 @@ export type QueryMembersOptions = {
   user_id_lte?: string;
 };
 
-export type SearchOptions = {
+export type SearchOptions = LimitOffsetSearchOptions | NextSearchOptions;
+
+export type LimitOffsetSearchOptions = {
   limit?: number;
   offset?: number;
 };
 
+export type NextSearchOptions = {
+  limit?: number;
+  next?: string;
+  sort?: SearchMessageSort;
+};
```

Comment:
```suggestion
export type LimitOffsetSearchOptions = {
  limit?: number;
  offset?: number;
};

export type NextSearchOptions = {
  limit?: number;
  next?: string;
  sort?: SearchMessageSort;
};

export type SearchOptions = LimitOffsetSearchOptions | NextSearchOptions;
```

Reply:
Defining before using ... makes it a little easier to read!!

Reply:
Also, do we really need separate `LimitOffsetSearchOptions` and  `NextSearchOptions`? Because anyways user can't switch between v1 or v2 search, right?

```
export type SearchOptions = {
  limit?: number;
  next?: string;
  sort?: SearchMessageSort;
};
```

Reply:
I thought it was better to keep them separate - A user should not be allowed to use offset with next or with sort.

Reply:
Ah my bad. I thought `sort` was the only difference there :) Thanks for clarification